internal/internal_task_pollers.go

```diff
@@ -252,17 +248,25 @@ func (bp *basePoller) doPoll(
 	var err error
 	var result interface{}
 
-	doneC := make(chan struct{})
 	ctx, cancel, _ := newChannelContext(context.Background(), featureFlags, chanTimeout(pollTaskServiceTimeOut))
 
```

Comment:
while you're here:
```suggestion
	ctx, cancel, _ := newChannelContext(context.Background(), featureFlags, chanTimeout(pollTaskServiceTimeOut))
	defer cancel()

```
and then I'd just drop the explicit `cancel()` code in the select case further down.
cancel is safe to call any number of times, and it's a bit more future-proof to do it immediately.